### ISSUE TITLE
Serializable class renamed in the executable.

### ISSUE THREAD

**Issue by AzuNoldus** [yck1509/ConfuserEx#519]
Hi,
I have a public class marked as serializable in the executable. I use rename protection and this class gets renamed whereas identical class in the class library is left unobfuscated. My application still works but the executable produces obfuscated XML files.
Is this intentional? I thought that all public serializable classes should not be confused.

**Comment by Rottweiler** [comment]
If I could get some example code of what is being obfuscated, I may be able to fix this issue.

**Comment by Rottweiler** [comment]
@AzuNoldus I made a patch and pushed to yck1509/ConfuserEx, if you want to get a preview of the changes (which enables xml serialization, you can get it here https://github.com/Rottweiler/ConfuserEx

**Comment by AzuNoldus** [comment]
@Rottweiler thanks for looking into the issue. Unfortunately your changes didn't solve the problem. I made a small test application which produces obfuscated settings.xml file. I hope this helps.

[ConfuserExample.zip](https://github.com/yck1509/ConfuserEx/files/524687/ConfuserExample.zip)

**Comment by Rottweiler** [comment]
@AzuNoldus You're right! I forgot to leave out renaming for property types. Pushed an update.

**Comment by AzuNoldus** [comment]
@Rottweiler Sorry for not responding earlier. Today I tested your latest patch and it seems that serialization is fixed. Thanks! But I have other problems with name protection. I have some public classes  and methods that are compiled into the executable and those are confused. I need to invoke those methods from javascript which cannot find them anymore. It seems that public methods are not confused inside of the class library (.dll). There a separate options for that (renPublic) Is this intentionally?
Btw, entity framework migrations are also confused. This makes it impossible to manually apply / revert migrations.

**Comment by Rottweiler** [comment]
I believe there is an attribute you can put on the things you don't want renamed! Like [DoNotRename], or similar

**Comment by AzuNoldus** [comment]
Yes, I am using now selective obfuscation to exclude things from being renamed. I was just wondering if the renaming of public things is done intentionally or that is actually a bug.

**Comment by Rottweiler** [comment]
I believe all things are meant to be renamed, except Serializable marked things

**Comment by AzuNoldus** [comment]
@Rottweiler I don't believe this is true. Public symbols are visible outside of the assembly. Think about the interfaces. If these get renamed nobody can interact with this assembly from outside anymore. There is a parameter in the [Name Protection](https://github.com/yck1509/ConfuserEx/wiki/Name-Protection) "renPublic" which by default is false:

_"renPublic: This parameter is a boolean value, indicates whether ConfuserEx should rename the symbols even if the item is visible outside the assembly. Default is false."_
